x-pack/plugins/cloud_integrations/cloud_chat/public/index.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { PluginInitializerContext } from '@kbn/core/public';
+import { CloudChatPlugin } from './plugin';
+
+export function plugin(initializerContext: PluginInitializerContext) {
+  return new CloudChatPlugin(initializerContext);
+}
+
+export { Chat } from './components';
```

Comment:
I am not aware of when and why that decision was made. I agree that this could have been exposed as part of the plugin's contract and avoid the `CloudContextProvider`'s part